src/browser/modules/Editor/Monaco.tsx

```diff
@@ -76,410 +65,379 @@ const MonacoStyleWrapper = styled.div`
+        },
+        (response: { result: QueryResult; success?: boolean }) => {
+          if (
+            response.success === true &&
+            response.result.summary.notifications.length > 0
+          ) {
+            editor.setModelMarkers(model, this.getMonacoId(), [
+              ...editor.getModelMarkers({ owner: this.getMonacoId() }),
+              ...response.result.summary.notifications.map(
+                ({ description, position, title }) => {
+                  const line = 'line' in position ? position.line : 0
+                  const column = 'column' in position ? position.column : 0
+                  return {
+                    startLineNumber: statementLineNumber + line,
+                    startColumn:
+                      statement.start.column +
+                      (line === 1
+                        ? column - EXPLAIN_QUERY_PREFIX_LENGTH
+                        : column),
+                    endLineNumber: statement.stop.line,
+                    endColumn: statement.stop.column + 2,
+                    message: title + '\n\n' + description,
+                    severity: MarkerSeverity.Warning
+                  }
+                }
+              )
+            ])
+          }
+        }
+      )
+    })
+  }
 
-    const updateGutterCharWidth = (dbName: string) => {
-      editorRef.current?.updateOptions({
-        lineNumbersMinChars:
-          dbName.length && !isMultiLine() ? dbName.length * 1.3 : 2
+  componentDidMount = (): void => {
+    this.container = document.getElementById(this.getMonacoId()) ?? undefined
+    if (!this.container) return
+
+    this.editor = editor.create(this.container, {
+      autoClosingOvertype: 'always',
+      contextmenu: false,
+      cursorStyle: 'block',
+      fontFamily: '"Fira Code", Monaco, "Courier New", Terminal, monospace',
+      fontLigatures: this.props.fontLigatures,
+      fontSize: 17,
+      fontWeight: '400',
+      hideCursorInOverviewRuler: true,
+      language: 'cypher',
+      lightbulb: { enabled: false },
+      lineHeight: 23,
+      lineNumbers: (line: number) =>
+        this.isMultiLine() ? line.toString() : `${this.props.useDb || ''}$`,
+      links: false,
+      minimap: { enabled: false },
+      overviewRulerBorder: false,
+      overviewRulerLanes: 0,
+      quickSuggestions: false,
+      renderLineHighlight: 'none',
+      scrollbar: {
+        alwaysConsumeMouseWheel: false,
+        useShadows: false
+      },
+      scrollBeyondLastColumn: 0,
+      scrollBeyondLastLine: false,
+      selectionHighlight: false,
+      value: this.props.value,
+      wordWrap: 'on',
+      wrappingStrategy: 'advanced'
+    })
+
+    this.editor.addCommand(
+      KeyCode.Enter,
+      () => {
+        this.isMultiLine() ? this.newLine() : this.execute()
+      },
+      '!suggestWidgetVisible && !findWidgetVisible'
+    )
+    this.editor.addCommand(
+      KeyCode.UpArrow,
+      this.handleUp,
```

Comment:
It'd be nice to have more consistent naming of the command callbacks, but that's a nit.